src/App.tsx

```diff
@@ -44,6 +44,16 @@ function App() {
     textarea
   );
   const setTextToParseThrottle = useThrottleCallback(setTextToParse, 2);
+  const inputRef = useRef({} as HTMLTextAreaElement);
+  const [currentPosition, setCurrentPosition] = useState({
+    selectionStart: textarea.length,
```

Comment:
We're storing selectionStart and selectionEnd here, but we only really need one, since we're setting the cursor position rather than selecting a group of characters